lib/antiCheat.ts

```diff
@@ -44,6 +45,37 @@ exports.calculateFindItCheatScore = async (challenge: Challenge) => { // TODO Co
   const timestamp = new Date()
   let timeFactor = 0.001
   timeFactor *= (challenge.key === 'scoreBoardChallenge' && config.get('hackingInstructor.isEnabled') ? 0.5 : 1)
+
+  // checks if identicalChallenges is already fetched
+  if (identicalChallenges.length === 0) {
+    identicalChallenges = await retrieveIdenticalChallenges()
+  }
+
+  // identicalChallenges[identicalIndex] will store an array of challenges identical to challenge.key
+  let identicalIndex = -1
+  identicalChallenges.forEach((row, i) => {
+    row.forEach((element) => {
+      if (element === challenge.key) {
+        identicalIndex = i
+      }
+    })
+  })
```

Comment:
Getting the index out doesn't seem to be required here as we primarly care about the group so we can just get it out directly.

```suggestion
  const identicalChallengeGroupForSolvedChallenge = identicalChallengeGroups.find(
    (identicalChallengeGroup) => identicalChallengeGroups.has(challenge.key)
  )
```